test/links.c

```diff
@@ -2117,6 +2117,22 @@ test_deprec(hid_t fapl, bool new_format)
     if (H5Gmove2(file_id, "group2", group1_id, "moved_group2") < 0)
         FAIL_STACK_ERROR;
 
+    /* Test passing in invalid ID */
+    H5E_BEGIN_TRY
+    {
+        if (H5Gmove2(0, "group2", group1_id, "moved_group2") >= 0)
```

Comment:
I would make an explicit bad_id = H5I_INVALID_HID variable and pass that in, instead of using a normal ID that happens to be set to H5I_INVALID_HID. If someone reworks the test, that might get set.

Reply:
Actually, this test is meant to detect 0 being passed in, like in the user's sample program.

Reply:
Nevermind, 0 is the case of H5L_SAME_LOC.  I'll have to redo this.